widgetssdk/src/main/java/com/glia/widgets/entrywidget/adapter/EntryWidgetAuthenticatedContactViewHolder.kt

```diff
@@ -0,0 +1,68 @@
+package com.glia.widgets.entrywidget.adapter
+
+import android.view.View
+import androidx.core.view.isVisible
+import com.glia.widgets.R
+import com.glia.widgets.core.secureconversations.domain.ObserveUnreadMessagesCountUseCase
+import com.glia.widgets.databinding.EntryWidgetAuthenticatedContactBinding
+import com.glia.widgets.entrywidget.EntryWidgetContract
+import com.glia.widgets.helper.setLocaleHint
+import com.glia.widgets.helper.setLocaleText
+import com.glia.widgets.view.unifiedui.applyImageColorTheme
+import com.glia.widgets.view.unifiedui.applyLayerTheme
+import com.glia.widgets.view.unifiedui.applyTextTheme
+import com.glia.widgets.view.unifiedui.theme.entrywidget.MediaTypeItemTheme
+import io.reactivex.rxjava3.disposables.Disposable
+import java.util.Locale
+
+internal class EntryWidgetAuthenticatedContactViewHolder(
+    private val binding: EntryWidgetAuthenticatedContactBinding,
+    itemTheme: MediaTypeItemTheme?,
+    private val observeUnreadMessagesCountUseCase: ObserveUnreadMessagesCountUseCase
```

Comment:
I would strongly recommend not passing the use case through EntryWidgetView → Adapter → ViewHolder. Instead, we can use it inside the controller and manipulate items before passing them to the adapter, as we usually do.
This way, we will:

- Keep business logic out of the view layer
- Avoid the complexity of passing use cases through multiple layers 
- Avoid handling of the RecyclerView item view lifecycle to dispose observables, etc.

Reply:
Thank you. 👍 
Please have a look 619d518fac903575fde6e24f9cbb9e471c6fc81a 😉